YDotNet/Document/Types/Branches/Branch.cs

```diff
@@ -36,60 +53,48 @@ protected Branch(nint handle)
     /// </remarks>
     /// <param name="action">The callback to be executed when a <see cref="Transaction" /> is committed.</param>
     /// <returns>The subscription for the event. It may be used to unsubscribe later.</returns>
-    public EventSubscription ObserveDeep(Action<IEnumerable<EventBranch>> action)
+    public IDisposable ObserveDeep(Action<IEnumerable<EventBranch>> action)
     {
-        var subscriptionId = BranchChannel.ObserveDeep(
-            Handle,
-            nint.Zero,
-            (_, length, eventsHandle) =>
-            {
-                var events = MemoryReader.TryReadIntPtrArray(eventsHandle, length, size: 24)!
-                    .Select(x => new EventBranch(x))
-                    .ToArray();
-
-                action(events);
-            });
-
-        return new EventSubscription(subscriptionId);
+        return onDeep.Subscribe(action);
     }
 
     /// <summary>
-    ///     Unsubscribes a callback function, represented by an <see cref="EventSubscription" /> instance, for changes
-    ///     performed within <see cref="Branch" /> scope.
+    ///     Starts a new read-write <see cref="Transaction" /> on this <see cref="Branch" /> instance.
     /// </summary>
-    /// <param name="subscription">The subscription that represents the callback function to be unobserved.</param>
-    public void UnobserveDeep(EventSubscription subscription)
+    /// <returns>The <see cref="Transaction" /> to perform operations in the document.</returns>
+    /// <exception cref="YDotNetException">Another write transaction has been created and not commited yet.</exception>
+    public Transaction WriteTransaction()
     {
-        BranchChannel.UnobserveDeep(Handle, subscription.Id);
+        ThrowIfDisposed();
+
+        var handle = BranchChannel.WriteTransaction(Handle);
+
+        if (handle == nint.Zero)
+        {
+            ThrowHelper.PendingTransaction();
```

Comment:
Okay for me. But the output decision is because I don't trust that the value of an output are still accessible later (e.g. if the output is stored somewhere after an event).

Reply:
Yeah, I understand that part and I think it's fine to keep the eager-build approach on `Output` for now for reliability.